### PR TITLE
feat(op-dispute-mon): CI Dispute Monitor Wiring

### PR DESCRIPTION
**Pull request by refcell**
**Description**

Wires the `op-dispute-mon` into ci and sets up docker.

**Metadata**

Fixes https://github.com/ethereum-optimism/client-pod/issues/531

### CODE REVIEW

**Comment by refcell** [comment]
Current dependencies on/for this PR:
* `develop`
  * **PR #9359** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/optimism/9359?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #9365** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/optimism/9365?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/ethereum-optimism/optimism/9365?utm_source=stack-comment).

File: op-dispute-mon/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+FROM alpine:3.18
+
+# Make the bundled op-program the default cannon server
+COPY --from=builder /usr/local/bin/op-program /usr/local/bin/op-program
+ENV OP_DISPUTE_MON_CANNON_SERVER /usr/local/bin/op-program
+
+# Make the bundled cannon the default cannon executable
+COPY --from=builder /usr/local/bin/cannon /usr/local/bin/cannon
+ENV OP_DISPUTE_MON_CANNON_BIN /usr/local/bin/cannon
```
Comment (ajsutton) [inline op-dispute-mon/Dockerfile]: We won't need op-program or cannon in the dispute monitor. Its purely lightweight and never actually runs these.

File: op-dispute-mon/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,17 @@
+ARG OP_STACK_GO_BUILDER=us-docker.pkg.dev/oplabs-tools-artifacts/images/op-stack-go:latest
```
Comment (ajsutton) [inline op-dispute-mon/Dockerfile:1]: I think we need to update the `docker-bake.hcl` to have a `op-dispute-mon` target as well, similar to the `op-challenger` target it has. That will fix the `error: failed to find target op-dispute-mon` failure we're getting.

**Review by ajsutton — COMMENTED** [review]
Looking good - just need to get the docker build working.

**Comment by ajsutton** [comment]
We should add `op-dispute-mon` to the list of targets in `golang-docker` too in the top level `Makefile`.  But I'm not sure what needs to change to get op-dispute-mon included in the docker build context to fix the not found error cd is getting currently.

**Review by ajsutton — APPROVED** [review]
LGTM.

**Pull request merged**